src/str.rs

```diff
@@ -34,62 +35,46 @@ fn find_char_midpoint(chars: &str) -> usize {
 
 
 /// Parallel extensions for strings.
-///
-/// Implementing this trait is not permitted outside of `rayon`.
-pub trait ParallelString {
-    private_decl!{}
-
+pub trait ParallelString: Borrow<str> {
```

Comment:
The linked internals thread was more about calling pub-in-private a negative pattern, to the point that it might deserve a warning or error - especially for cases like this that couldn't just use pub(restricted).

I didn't consider `Deref`, but I see how that's an even tighter choice since there can be only one `Target` for a given type.  I'm in favor.